### PR TITLE
Java libary updates and API improvements.

### LINKED ISSUE #27: Java VaultClient constructor should take in interfaces, not concrete classes
The VaultClient consturctor takes following parameters
- com.amazonaws.services.s3.AmazonS3Client
- com.amazonaws.services.kms.AWSKMSClient.AWSKMSClient

https://github.com/NitorCreations/vault/blob/master/java/src/main/java/com/nitorcreations/vault/VaultClient.java#L41

Both of these are concrete classes. Better would be to take the interfaces as parameters. For example AWS sdk funktions typically return the interfaces, not the concrete classes.
- com.amazonaws.services.s3.AmazonS3
- com.amazonaws.services.kms.AWSKMS

### PR DESCRIPTION
**Pull request by jsyrjala**
Fixes #27.

Update libraries.

### CODE REVIEW

File: java/src/main/java/com/nitorcreations/vault/VaultClient.java [hunk 1]
```diff
@@ -37,6 +39,11 @@
   private static final String VALUE_OBJECT_NAME_FORMAT = "%s.%s";
   private static final String KEY_OBJECT_NAME_FORMAT = "%s.key";
 
+  @Deprecated
+  public VaultClient(AmazonS3Client s3, AWSKMSClient kms, String bucketName, String vaultKey) {
```
Comment (jsyrjala) [inline java/src/main/java/com/nitorcreations/vault/VaultClient.java]: The new constructor works also when calling with concrete classes.

Comment (psiniemi) [reply java/src/main/java/com/nitorcreations/vault/VaultClient.java]: Yes, but having the deprecated constructor makes it clear that people should change their code

**Comment by Jalle19** [comment]
Can you make a release that includes these changes? `0.14` is the latest in Maven Central and it doesn't have these improvements. There's no `aws-vault-2` package despite what the README says (https://github.com/NitorCreations/vault/blob/master/java-2/README.md).

**Comment by psiniemi** [comment]
Done, 0.15 of both released.

**Pull request merged**